src/peer-store/address-book.js

```diff
@@ -109,12 +218,14 @@ class AddressBook extends Book {
 
     const addresses = this._toAddresses(multiaddrs)
     const id = peerId.toB58String()
-    const rec = this.data.get(id)
+
+    const entry = this.data.get(id) || {}
+    const rec = entry.addresses
 
     // Add recorded uniquely to the new array (Union)
-    rec && rec.forEach((mi) => {
-      if (!addresses.find(r => r.multiaddr.equals(mi.multiaddr))) {
-        addresses.push(mi)
+    rec && rec.forEach((addr) => {
```

Comment:
Combine with the empty array set above.
```suggestion
    rec.forEach((addr) => {
```